src/EventSubscriber/PostRowSave.php

```diff
@@ -56,9 +68,42 @@ public static function getSubscribedEvents() {
    * @throws \Drupal\migrate\MigrateSkipRowException
    */
   public function onMigratePostRowSave(MigratePostRowSaveEvent $event) {
-    // $migration = $event->getMigration();
-    // $row = $event->getRow();
-    // $nids = $event->getDestinationIdValues();
+    $row = $event->getRow();
+    $migration_tools_settings = $row->getSourceProperty('migration_tools');
+
+    if (!empty($migration_tools_settings)) {
+      // Create redirects if enabled.
+      // @todo Current only supports 1st migration_tools array entry.
+      $source_type = $migration_tools_settings[0]['source_type'];
+      $source = $migration_tools_settings[0]['source'];
+      $create_redirects = isset($migration_tools_settings[0]['create_redirects']) ? $migration_tools_settings[0]['create_redirects'] : FALSE;
+
+      if ($source_type == 'url' && !empty($source) && $create_redirects) {
+        $preserve_query_params = isset($migration_tools_settings[0]['redirect_preserve_query_params']) ? $migration_tools_settings[0]['redirect_preserve_query_params'] : FALSE;
+        $source_url = $row->getSourceProperty($source);
+        $nids = $event->getDestinationIdValues();
+        $source_url_pieces = parse_url($source_url);
+        $source_path = ltrim($source_url_pieces['path'], '/');
+        $source_query = [];
+
+        if ($preserve_query_params) {
+          $source_query = isset($source_url_pieces['query']) ? $source_url_pieces['query'] : [];
+        }
+        $nid = $nids[0];
+
+        // Check if redirect already exists first before creating.
```

Comment:
We need a way to name space the redirect $source_path before this line.   This is essential for avoiding redirect collisions on both single site migrations and absolutely essential for avoiding multiple sites being federated into one.

Reply:
For the bigger picture of this, see slides 11-13  This will take you to slide 11 ->
https://docs.google.com/presentation/d/1LhLJoTPyauyYETOYukC2enemnEJKOJ5SA8deJKLQu2I/pub?start=false&loop=false&delayms=60000&slide=id.g20d47874fc_0_53

Reply:
Done